### PR TITLE
Bug 5831/v1

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/5831

### CODE REVIEW

File: src/util-device.h [hunk 1]
```diff
@@ -49,6 +49,7 @@ typedef struct {
 typedef struct LiveDevice_ {
     char *dev;  /**< the device (e.g. "eth0") */
     char dev_short[MAX_DEVNAME + 1];
+    uint8_t mtu; /* MTU of the device */
```
Comment (victorjulien) [inline src/util-device.h:52]: incorrect type. Common mtu is 1500, we also have 9000. Should be at least uint16_t, but perhaps just make it `int`

File: src/util-ioctl.c [hunk 2]
```diff
@@ -124,11 +124,12 @@ int GetIfaceMTU(const char *pcap_dev)
...
  * majorant to be sure avoid the cost of dynamic allocation.
  *
- * \param Name of a network interface
+ * \param LiveDevice object
  * \retval 0 in case of error
  */
-int GetIfaceMaxPacketSize(const char *pcap_dev)
+int GetIfaceMaxPacketSize(LiveDevice *ld)
 {
+    const char *pcap_dev = ld->dev;
```
Comment (victorjulien) [inline src/util-ioctl.c:132]: as a minor unrelated cleanup, lets rename this to just `dev`. It's not specific to pcap.

**Review by victorjulien — CHANGES_REQUESTED** [review]

File: src/source-af-packet.c [hunk 3]
```diff
@@ -527,12 +526,14 @@ static TmEcode AFPPeersListAdd(AFPThreadVars *ptv)
                 continue;
             peer->peer = pitem;
             pitem->peer = peer;
-            mtu = GetIfaceMTU(ptv->iface);
-            out_mtu = GetIfaceMTU(ptv->out_iface);
-            if (mtu != out_mtu) {
+
+            LiveDevice *iface = LiveGetDevice(ptv->iface);
```
[the remark this answers is not in the document]
Comment (inashivb) [reply src/source-af-packet.c:530]: oh yes. unnecessary call. will fix

File: src/util-device.h (hunk 1 above)
Comment (victorjulien) [reply src/util-device.h:52]: even 1500 wouldn't have fitted in u8

Comment (inashivb) [reply src/util-device.h:52]: indeed 🤦🏽
